### PR TITLE
Add thread safe method of querying BigCommerce resources.

### LINKED ISSUE #98: Is the new gem thread safe?
If we are using a sidekiq with multiple threads and do
````
Bigcommerce.configure do |config|
  config.store_hash = ENV['BC_STORE_HASH']
  config.client_id = ENV['BC_CLIENT_ID']
  config.access_token = ENV['BC_ACCESS_TOKEN']
end

# List orders
@orders = Bigcommerce::Order.all
````
in each worker, and they are executing at the same time, will `Bigcommerce` be isolated to the correct client in each worker?

### PR DESCRIPTION
**Pull request by pedelman**
Example usage

```rb
# Thread safe
conn = Bigcommerce::Connection.build(
  Bigcommerce::Config.new(
    store_hash: ENV['BC_STORE_HASH'], 
    client_id: ENV['BC_CLIENT_ID'], 
    access_token: ENV['BC_ACCESS_TOKEN']
  )
)

Bigcommerce::System.time(connenction: conn)
=> #<Bigcommerce::System time=1466546702>

Bigcommerce::System.raw_request(:get, '/time', connection: conn)
=> #<Faraday::Response:0x007fd4a4063170 ... >>
```

```rb
# Not thread safe
Bigcommerce.configure do |config|
  config.store_hash = ENV['BC_STORE_HASH']
  config.client_id = ENV['BC_CLIENT_ID']
  config.access_token = ENV['BC_ACCESS_TOKEN']
end

Bigcommerce::System.time
=> #<Bigcommerce::System time=1466546702>

Bigcommerce::System.raw_request(:get, '/time')
=> #<Faraday::Response:0x007fd4a4063170 ... >>
```

Fixes #98

### CODE REVIEW

**Comment by pedelman** [comment]
ping @mattolson @saurabh-g @jason0415 @gregory

File: lib/bigcommerce/connection.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module Bigcommerce
+  class Connection
+    attr_reader :connection
```
Comment (mattolson) [inline lib/bigcommerce/connection.rb]: What is this for?

File: lib/bigcommerce/connection.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Bigcommerce
+  class Connection
```
Comment (mattolson) [inline lib/bigcommerce/connection.rb]: This could just be a module, there is no state here

**Comment by gregory** [comment]
had a quick high level overview but not tested. 
Overall looks good!
One thing i'm not really a fan of is [tempering with parameters](https://github.com/bigcommerce/bigcommerce-api-ruby/pull/123/files#diff-91ed787e8db6ee761397fc63d3f9e36bR67) 
I do believe that passing connection objects is the way to go to solve thread safety but also have the ability to query multiple stores.

**Comment by gregory** [comment]
Also, don't you feel like it's kind of a "hack" to [allow to pass parameters](https://github.com/bigcommerce/bigcommerce-api-ruby/pull/123/files#diff-0ac07d779ddd8d8d4f33cc01a60d38cfR36), just to be able to sneak in the connection object? :) 

What i'm saying is that you allow me to do this: `Bigcommerce::System.time(foo: 'bar')` but this is just so you can pass in a connection: `Bigcommerce::System.time(connenction: conn)`

**Comment by mattolson** [comment]
👍

**Pull request merged**